source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1401,6 +1430,27 @@ void Filter::addDynamicMetadata(const ProcessorState& state, ProcessingRequest&
   auto* cb = state.callbacks();
   envoy::config::core::v3::Metadata forwarding_metadata;
 
+  // Forward cluster metadata if so configured.
+  const auto& cluster_info = cb->streamInfo().upstreamClusterInfo();
```

Comment:
I guess these change is fine even if the ext_proc filter is in the upstream filter chain. Please confirm!